subsys/usb/usb_device.c

```diff
@@ -249,10 +250,16 @@ static void usb_data_to_host(u16_t len)
 		 * last chunk is wMaxPacketSize long, to indicate the last
 		 * packet.
 		 */
-		if (!usb_dev.data_buf_residue && chunk == USB_MAX_CTRL_MPS
-		    && len > chunk) {
-			usb_dev.zlp_flag = true;
+		if (!usb_dev.data_buf_residue && len > usb_dev.data_buf_len) {
+			/* Send less data as requested during the Setup stage */
+			if (!(usb_dev.data_buf_len % USB_MAX_CTRL_MPS)) {
+				/* Transfers a zero-length packet */
+				LOG_ERR("ZLP, requested %u , length %u ",
```

Comment:
I don't think that LOG_ERR is the correct level here. Sending zlp isn't an error.

Reply:
o yes, I used it for debugging, will fix, thanks